File: 151051/Maliuk/ipr/src/main/java/ru/bsuir/dto/request/CreatorRequestTo.java
package ru.bsuir.dto.request;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Data;

@Data
public class CreatorRequestTo {

    private Long id;

    @NotBlank
    @Size(min = 2, max = 64)
    String login;

    @NotBlank
    @Size(min = 2, max = 128)
    String password;

    @NotBlank
    @Size(min = 2, max = 64)
    String firstname;

    @NotBlank
    @Size(min = 2, max = 64)
    String lastname;
}

File: 251002/Nikonorova/lab1_rest/DTOs/CreatorRequestTo.cs
using System.ComponentModel.DataAnnotations;

namespace distcomp.DTOs
{
    public class CreatorRequestTo
    {
        public long Id { get; set; } 

        [Required]
        [StringLength(64, MinimumLength = 2)]
        public string Login { get; set; } 

        [Required]
        [StringLength(128, MinimumLength = 8)]
        public string Password { get; set; } 

        [Required]
        [StringLength(64, MinimumLength = 2)]
        public string Firstname { get; set; } 

        [Required]
        [StringLength(64, MinimumLength = 2)]
        public string Lastname { get; set; } 
    }
}
